.github/workflows/kotlin-release.yml

```diff
@@ -0,0 +1,131 @@
+        required: true
+      SONATYPE_ACCOUNT_GPG_KEY_ID:
+        required: true
+      SONATYPE_ACCOUNT_GPG_KEY_PASSWORD:
+        required: true
+
+jobs:
+  release-check:
+    name: Check if version is published
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: ${{ inputs.working-directory }}
+    outputs:
+      current-version: ${{ steps.check.outputs.current-version }}
+      published: ${{ steps.check.outputs.published }}
+      previous-version: ${{ steps.check.outputs.previous-version }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Read version from build.gradle.kts
+        id: read-version
+        run: |
+          version=$(grep 'coordinates.*auth.*' library/build.gradle.kts | sed 's/.*"\([^"]*\)".*/\1/')
+          echo "current-version=$version" >> $GITHUB_OUTPUT
+
+      - name: Check if tag already exists
+        id: check
+        run: |
+          current="${{ steps.read-version.outputs.current-version }}"
+          previous=$(git describe --tags --abbrev=0 2>/dev/null || echo "0.0.0")
+
+          if git rev-parse "v$current" >/dev/null 2>&1; then
+            published=true
+          else
+            published=false
+          fi
+
+          echo "published=$published" >> $GITHUB_OUTPUT
+          echo "previous-version=$previous" >> $GITHUB_OUTPUT
+          echo "current-version=$current" >> $GITHUB_OUTPUT
+
+  release_to_github_and_maven:
+    needs: release-check
+    if: ${{ needs.release-check.outputs.published == 'false' }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      actions: write
+    defaults:
+      run:
+        shell: bash
+        working-directory: ${{ inputs.working-directory }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: "temurin"
+          java-version: 17
+
+      - name: Setup Android SDK
+        uses: android-actions/setup-android@v3
+
+      - name: Build with Gradle
+        run: ./gradlew build
+
+      # - name: Run tests
+      #   run: ./gradlew test
+
+      - name: Extract changelog for version
+        run: |
+          if [ -f "CHANGELOG.md" ]; then
+            awk '/# ${{needs.release-check.outputs.current-version }}/{f=1;next}/# ${{needs.release-check.outputs.previous-version}}/{f=0} f' CHANGELOG.md > changelog_for_version.md
+          else
+            echo "Release ${{ needs.release-check.outputs.current-version }}" > changelog_for_version.md
+          fi
+          cat changelog_for_version.md
+
+      - name: Create git tag
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+          git tag v${{ needs.release-check.outputs.current-version }}
+          git push origin v${{ needs.release-check.outputs.current-version }}
```

Comment:
Does the tag need a prefix as well?

Reply:
The tags are good/easy way for folks to pull down a tagged release (if they wanted to). And I believe the logic is currently checking the tags to determine if the current version has already been published.